imap_processing/glows/l0/decom_glows.py

```diff
@@ -49,39 +49,28 @@ def decom_packets(
         f"{imap_module_directory}/glows/packet_definitions/GLX_COMBINED.xml"
     )
 
-    packet_definition = xtcedef.XtcePacketDefinition(xtce_document)
-    glows_parser = parser.PacketParser(packet_definition)
+    packet_definition = definitions.XtcePacketDefinition(xtce_document)
 
     histdata = []
     dedata = []
 
     filename = packet_file_path.name
 
     with open(packet_file_path, "rb") as binary_data:
-        glows_packets = glows_parser.generator(binary_data)
+        glows_packets = packet_definition.packet_generator(binary_data)
 
         for packet in glows_packets:
-            apid = packet.header["PKT_APID"].derived_value
+            apid = packet.header["PKT_APID"]
             # Do something with the packet data
             if apid == GlowsParams.HIST_APID.value:
-                values = [
-                    item.derived_value
-                    if item.derived_value is not None
-                    else item.raw_value
-                    for item in packet.data.values()
-                ]
+                values = [item.raw_value for item in packet.user_data.values()]
```

Comment:
Before this was getting the derived value if it is not None. The `if item.derived_value is not None else item.raw_value` is taken care of for you by space_packet_parser now.

I'm pretty sure this is the equivalent to before now:
```suggestion
                values = list(packet.user_data.values())
```

Reply:
@maxinelasp what do you think here? I thought same as Greg but you helped me debug some of GLOWS' error and I saw that you had to get `raw_value`. Is that right?

Reply:
Yes, when I printed out the types here it was all `IntParameter` (etc) instead of `int`.

Reply:
```python
class IntParameter(int):
```
`IntParameter is a subclass of `int` so if you type it as `int` `IntParameter` should be acceptable as well? Alternatively, could you type it as `IntParameter` where this is needed?

`item.raw_value` is very different from `item`.

`item.derived_value` from v4.x is the same as `item` in v5.x